OpenEdXMobile/src/main/java/org/edx/mobile/util/CalendarUtils.kt

```diff
@@ -183,6 +185,41 @@ object CalendarUtils {
         )
     }
 
+    /**
+     * Method to compare the calendar events with course dates
+     * @return  true if the events are the same as calendar dates otherwise false
+     */
+    fun compareEvents(context: Context, calendarId: Long, courseDateBlocks: List<CourseDateBlock>): Boolean {
+        val cursor = getCalendarEvents(context, calendarId)
+        val datesList = ArrayList(courseDateBlocks)
```

Comment:
I don't think so we need to create a new reference for the same list.

Reply:
I created this copy to optimize the loop complexity